test/UnitTest/TestingControllers/DataControllerTests.cs

```diff
@@ -65,13 +66,34 @@ public async void Post_NewData_Ok()
             client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", PrincipalUtil.GetToken(1337, 1337, 3));
             HttpResponseMessage response = await client.PostAsync($"{dataPathWithData}?dataType=default", content);
 
-            if (response.StatusCode.Equals(HttpStatusCode.InternalServerError))
-            {
-                string serverContent = await response.Content.ReadAsStringAsync();
-                Assert.Equal("Hei", serverContent);
-            }
+            Assert.Equal(HttpStatusCode.Created, response.StatusCode);
+        }
+
+        [Fact]
+        public async void Post_NewDataThatRequiresFileScan_Ok()
+        {
+            // Arrange
+            string dataPathWithData = $"{_versionPrefix}/instances/1337/bc19107c-508f-48d9-bcd7-54ffec905306/data";
+            HttpContent content = new StringContent("This is a blob file");
+
+            Mock<IFileScanQueueClient> fileScanMock = new Mock<IFileScanQueueClient>();
```

Comment:
any reasoning behind mocking the queueClient rather than the dataService that the controller has a dependency on? Wouldn't it be cleaner to mock the immediate dependency ?

Reply:
These tests already include a full in memory http service with an almost complete middleware. In my opinion not very clean to begin with. 

Whenever I use the in memory http service I like to mock as little as possible, or as late as possible. This gives them "deeper" integrations.  This is different from unit tests were I like to mock immediate dependencies.